src/workerd/api/node/node.h

```diff
@@ -74,7 +74,8 @@ constexpr bool isExperimentalNodeJsCompatModule(kj::StringPtr name) {
 constexpr bool isNodeHttpModule(kj::StringPtr name) {
   return name == "node:http"_kj || name == "node:_http_common"_kj ||
       name == "node:_http_outgoing"_kj || name == "node:_http_client"_kj ||
-      name == "node:_http_incoming"_kj || name == "node:_http_agent"_kj || name == "node:https"_kj;
+      name == "node:_http_incoming"_kj || name == "node:_http_agent"_kj ||
+      name == "node:https"_kj || name == "node:_http_server"_kj;
```

Comment:
Let's put the server under a separate compat flag that we can later implicitly switch on using `implied_by`. That will allow us to safely roll it out experimentally independently of the already completed client side APIs.

Reply:
How so? Just have `createServer(...)` throw an error if the compat flag is not set. The method can still be exposed.

Reply:
That's possible but even throwing an error of not implemented is still a (small but) breaking change, which is fine to dismiss. I'm just raising concern.

Reply:
Breaking how? The polyfills currently throw an error for these? could we not just throw the same error?

Reply:
> Breaking how? The polyfills currently throw an error for these? could we not just throw the same error?

Customers that have no_nodejs_v2_compat or an older compat flag will see a different behavior.

Reply:
Ok but that should be fine.